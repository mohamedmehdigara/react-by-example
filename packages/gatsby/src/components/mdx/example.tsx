import React, { FC, ReactNode, useCallback, useState } from 'react'
import styled from 'styled-components'
import { CodeSandboxIcon, ResetIcon } from '../icons'

const Render = styled.div`
  display: flex;
  padding: 1em;
  border: 1px solid var(--app-foreground);
  border-radius: var(--app-border-radius);
  align-items: center;
  justify-content: center;
`
const FlexRight = styled.div`
  display: flex;
  justify-content: flex-end;
`

const CodeSandboxLink = styled.a`
  display: flex;
`

const ResetButton = styled.button`
  display: flex;
  margin: 0;
  padding: 0;
  color: var(--app-foreground);
  background: none;
  border: none;
  cursor: pointer;

  :focus,
  :active {
    color: var(--app-cyan);
    outline: 2px solid var(--app-cyan);
  }
`

export interface ExampleProps {
  code?: ReactNode
  render?: ReactNode
  link?: string
}

export const Example: FC<ExampleProps> = ({ code, render, link }) => {
  const [resetKey, setResetKey] = useState(Date.now())
  const triggerReset = useCallback(() => {
    setResetKey(Date.now())
  }, [])
  return (
    <>
      {code}
      <Render key={resetKey}>
        <div>{render}</div>
      </Render>
      {link && (
        <FlexRight>
          <ResetButton
            type="button"
            title="reset example"
            onClick={triggerReset}
          >
            <ResetIcon />
          </ResetButton>
          <CodeSandboxLink
            target="_blank"
            rel="noreferrer"
            href={link}
            title="edit on CodeSandbox"
          >
            <CodeSandboxIcon />
          </CodeSandboxLink>
        </FlexRight>
      )}
    </>
  )
}
