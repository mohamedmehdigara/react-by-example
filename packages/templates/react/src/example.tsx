import React, { FC, ReactElement } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default (): ReactElement => <Simple />
