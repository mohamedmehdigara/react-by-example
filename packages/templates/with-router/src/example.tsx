import React, { FC } from 'react'

export const Simple: FC = () => <h1>Hello world!</h1>

export default <Simple />
