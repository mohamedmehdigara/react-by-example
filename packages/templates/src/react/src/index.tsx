import React from 'react'
import ReactDOM from 'react-dom'
import example from './example'

ReactDOM.render(<>{example}</>, document.getElementById('root'))
