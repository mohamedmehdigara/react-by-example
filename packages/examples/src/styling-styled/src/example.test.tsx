import { act, cleanup, render, RenderResult } from '@testing-library/react'
import React from 'react'
import { Styled } from './example'
import { SliderProps } from './slider'

const mockSliderOnChange = jest.fn<void, [number]>()
jest.mock('./slider.tsx', () => ({
  __esModule: true,
  Slider: ({ onChange }: SliderProps) => {
    mockSliderOnChange.mockImplementation(onChange)
    return <div id="slider" />
  },
}))

describe('Styled', () => {
  let component: RenderResult

  beforeEach(() => {
    component = render(<Styled />)
  })

  afterEach(() => {
    mockSliderOnChange.mockReset()
    cleanup()
  })

  it('should snapshot', () => {
    expect(component.container).toMatchSnapshot()
  })

  describe('when changing slider', () => {
    describe('max', () => {
      beforeEach(() => {
        act(() => {
          mockSliderOnChange(150)
        })
      })

      it('should update styles', () => {
        expect(component.container).toMatchSnapshot()
      })
    })

    describe('min', () => {
      beforeEach(() => {
        act(() => {
          mockSliderOnChange(75)
        })
      })

      it('should update styles', () => {
        expect(component.container).toMatchSnapshot()
      })
    })
  })
})
