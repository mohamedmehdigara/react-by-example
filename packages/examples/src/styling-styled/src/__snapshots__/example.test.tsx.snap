// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled should snapshot 1`] = `
.c0 {
  min-height: 250px;
  min-width: 250px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: dodgerblue;
  border-radius: 99999px;
  color: aliceblue;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: Arial,Helvetica,sans-serif;
  font-size: 1em;
  height: 10em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.5s cubic-bezier(0.68,-0.55,0.265,1.55);
  transition: all 0.5s cubic-bezier(0.68,-0.55,0.265,1.55);
  width: 10em;
}

<div>
  <div
    class="c0"
  >
    <div
      id="slider"
    />
    <div
      class="c1"
      font-size="1"
    >
      awesome circle
    </div>
  </div>
</div>
`;

exports[`Styled when changing slider max should update styles 1`] = `
.c0 {
  min-height: 250px;
  min-width: 250px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: dodgerblue;
  border-radius: 99999px;
  color: aliceblue;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: Arial,Helvetica,sans-serif;
  font-size: 1.5em;
  height: 10em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.5s cubic-bezier(0.68,-0.55,0.265,1.55);
  transition: all 0.5s cubic-bezier(0.68,-0.55,0.265,1.55);
  width: 10em;
}

<div>
  <div
    class="c0"
  >
    <div
      id="slider"
    />
    <div
      class="c1"
      font-size="1.5"
    >
      awesome circle
    </div>
  </div>
</div>
`;

exports[`Styled when changing slider min should update styles 1`] = `
.c0 {
  min-height: 250px;
  min-width: 250px;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: dodgerblue;
  border-radius: 99999px;
  color: aliceblue;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-family: Arial,Helvetica,sans-serif;
  font-size: 0.75em;
  height: 10em;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 0.5s cubic-bezier(0.68,-0.55,0.265,1.55);
  transition: all 0.5s cubic-bezier(0.68,-0.55,0.265,1.55);
  width: 10em;
}

<div>
  <div
    class="c0"
  >
    <div
      id="slider"
    />
    <div
      class="c1"
      font-size="0.75"
    >
      awesome circle
    </div>
  </div>
</div>
`;
